ldap/servers/slapd/back-ldbm/idl_new.c

```diff
@@ -626,17 +603,16 @@ idl_new_range_fetch(
     }
 
 error:
-    DBT_FREE_PAYLOAD(cur_key);
+    dblayer_value_free(be, &cur_key);
+    dblayer_bulk_free(&bulkdata);
     /* Close the cursor */
-    if (NULL != cursor) {
-        int ret2 = cursor->c_close(cursor);
-        if (ret2) {
-            ldbm_nasty("idl_new_range_fetch", filename, 3, ret2);
-            if (!ret) {
-                /* if cursor close returns DEADLOCK, we must bubble that up
-                   to the higher layers for retries */
-                ret = ret2;
-            }
+    ret2 = dblayer_cursor_op(&cursor, DBI_OP_CLOSE, NULL, NULL);
```

Comment:
What if dblayer_new_cursor failed and cursor is not allocated ?

Reply:
now cursor is usually the address of a local variable so it is always allocated.
 That is cursor->cur that is NULL in this scenario.
Hum should add a test in bdb_public_cursor_op and bdb_public_cursor_bulkop and return an error in that case